config/routes.rb

```diff
@@ -64,6 +64,14 @@ def redirect_with_vary(path)
                 tokens: 'oauth/tokens'
   end
 
+  namespace :oauth do
+    get 'userinfo', to: 'userinfo#show', defaults: { format: 'json' }
```

Comment:
If these have same options and only vary on accepted http method, look into `match ...,  via: [:get, :post]`

Reply:
Yeah, we could do that? I don't feel strongly about the way it is written.